packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -124,14 +141,15 @@ export class DefaultCrudRepository<T extends Entity, ID>
     // the juggler understands
     Object.entries(definition.properties).forEach(([key, value]) => {
       if (value.type === 'array' || value.type === Array) {
-        value = Object.assign({}, value, {type: [resolveType(value.itemType)]});
+        value = Object.assign({}, value, {
+          type: [value.itemType && this.resolvePropertyType(value.itemType)],
```

Comment:
We want juggler to see the new persisted model in the property definition, not a reference to LB4 model.